sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/_polling.py

```diff
@@ -0,0 +1,204 @@
+    :param deserialization_callback: A callback that takes a Response and return a deserialized object.
+                                     If a subclass of Model is given, this passes "deserialize" as callback.
+    :type deserialization_callback: callable or msrest.serialization.Model
+    :param polling_method: The polling strategy to adopt
+    :type polling_method: ~azure.core.polling.PollingMethod
+    """
+
+    def __init__(self, client, initial_response, deserialization_callback, polling_method):
+        # type: (Any, HttpResponse, DeserializationCallbackType, PollingMethod) -> None
+        # pylint: disable=super-init-not-called
+        self._client = client
+        self._response = initial_response
+        self._callbacks = []  # type: List[Callable]
+        self._polling_method = polling_method
+
+        # This implicit test avoids bringing in an explicit dependency on Model directly
+        try:
+            deserialization_callback = deserialization_callback.deserialize # type: ignore
+        except AttributeError:
+            pass
+
+        # Might raise a CloudError
+        self._polling_method.initialize(self._client, self._response, deserialization_callback)
+
+        # Prepare thread execution
+        self._thread = None
+        self._done = None
+        self._exception = None
+
+    # pylint: disable=arguments-differ
+    def result(self):
+        # type: (Optional[int]) -> Model
+        """Returns the result of the long running operation without blocking until its completion.
+
+        :returns: The deserialized resource of the long running operation,
+         if one is available.
+        :raises ~azure.core.exceptions.HttpResponseError: Server problem with the query.
+        """
+        return self._polling_method.resource()
+
+    @distributed_trace
+    def wait(self, timeout=None):
+        # type: (Optional[int]) -> None
+        """Wait on the long running operation for a specified length
+        of time. You can check if this call as ended with timeout with the
+        "done()" method.
+
+        :param int timeout: Period of time to wait for the long running
+         operation to complete (in seconds).
+        :raises ~azure.core.exceptions.HttpResponseError: Server problem with the query.
+        """
+
+        if not self._polling_method.finished():
+            self._done = threading.Event()
+            self._thread = threading.Thread(
+                target=with_current_context(self._start),
+                name="KeyVaultOperationPoller({})".format(uuid.uuid4()))
+            self._thread.daemon = True
+            self._thread.start()
+
+        if self._thread is None:
+            return
+        self._thread.join(timeout=timeout)
+        try:
+            # Let's handle possible None in forgiveness here
+            raise self._exception  # type: ignore
+        except TypeError: # Was None
+            pass
+
+
+
+class DeleteResourcePoller(PollingMethod):
```

Comment:
`DeleteResourcePollingMethod`?

Reply:
Agreed, switching them to *PollingMethod